### PR TITLE
Upgrade @Response to handle response body with specific conditions

### PR DESCRIPTION
**Pull request by dirong**

### CODE REVIEW

**Comment by almozavr** [comment]
It would be more strict and clear to define `SUCCESS` as -1, `ERROR` as -2 and think of 0 as unspecified. So, if value, min, max is 0 (read unspecified) compiler processor thinks it's success. Now I don't like `if(status>SUCCESS)` – it's unreadable.

**Comment by almozavr** [comment]
If (value, min, max is unspecified or value == success) add 200-300 checks
else if (value == error) add > 300 checks
else if (min > 0 || max > 0) add min,max checks

it relates to `HttpActionValidators` and `HttpHelpersGenerator` both

**Comment by dirong** [comment]
if you say that unspecified value should be as 0 so we have to remove constant `SUCCESS` as -1  be cause of there are two values for success status and we won't use this constant in the feature. 
unspecified == SUCCESS == 0
because of that I did SUCCESS as 0.

**Comment by dirong** [comment]
I realise what did u mean about. it's need to correctly check specified `value`.
the commit is updated. 
have I understood it right? 
check it please

File: janet-http/http-service-compiler/src/main/java/io/techery/janet/validation/ResponseValidator.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            if (annotation.max() < 0) {
+                return lessThanZero("@Response.max()", element);
+            }
+            if (annotation.min() < 0) {
+                return lessThanZero("@Response.min()", element);
+            }
+            if (annotation.value() != 0
+                    && (annotation.min() > 0 || annotation.max() > 0)) {
+                return new ValidationError("There is no possible to handle response with specific status code and range of statuses using arguments min() and max()", element);
+            }
```
Comment (VladSumtsov) [inline janet-http/http-service-compiler/src/main/java/io/techery/janet/validation/ResponseValidator.java]: It's forbidden to use min and max with value. But from your description, i can understand, that value isn't in range of min and max;

Comment (dirong) [inline janet-http/http-service-compiler/src/main/java/io/techery/janet/validation/ResponseValidator.java]: you are right. done
thanks

**Pull request merged**